File: src/BlockFarmEditor.Umbraco.Core/Models/BuilderModels/PageDefinition.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Umbraco.Cms.Core;
using Umbraco.Cms.Core.Models.PublishedContent;

namespace BlockFarmEditor.Umbraco.Core.Models.BuilderModels
{
    public class PageDefinition : IContainerDefinition
    {
        public const string GuidUnique = "64D51B94-10C1-4225-9329-B2A54D0E47CE";
        public IEnumerable<BlockDefinition<IPublishedElement>> Blocks { get; set; } = [];
        public Guid? ContentTypeKey => new(GuidUnique);

        public Guid Unique { get; set; } = new(GuidUnique);
    }
}

File: src/BlockFarmEditor.ClientScripts.RCL/models/PageDefinition.ts
import type { BlockDefinition } from "./BlockDefinition";
import type { IBuilderProperties } from "./IBuilderProperties";
import type { IContainerDefinition } from "./IContainerDefinition";

/**
 * PageDefinition class that implements IContainerDefinition
 */

export const PageDefinitionUnique = "64D51B94-10C1-4225-9329-B2A54D0E47CE";

export class PageDefinition implements IContainerDefinition {
    public identifier: string = "BlockFarmEditor";
    public unique: string = PageDefinitionUnique;
    public blocks: BlockDefinition<IBuilderProperties>[] = [];
}